sdk/ai/azure-ai-projects/azure/ai/projects/operations/_patch.py

```diff
@@ -1664,18 +1607,18 @@ def create_and_process_run(
                     logging.warning("No tool calls provided - cancelling run")
                     self.cancel_run(thread_id=thread_id, run_id=run.id)
                     break
+                # We need tool set only if we are executing local function. In case if
```

Comment:
For the create_and_process_run method, I think is better to replace tools parameter with toolset.

Otherwise, the code here might cause unexpected error, ie. if user defines tools (not toolset) for create_and_process_run method and then expect functions under tools definitions somehow to be executed, that would not happen. So, making create_and_process_run method requiring toolset for tools works more consistently.

Reply:
Fixed.